### ISSUE TITLE
[Engine] Make `screen-width` and `screen-height` define the in-game resolution

### ISSUE THREAD

**Issue by Wohlstand** [WohlSoft/PGE-Project#269]
The `engine.ini` has `screen-width` and `screen-height` options are unused yet internally. Make the sure there are will change in-game screen resolution when launching the game engine with choosing the config pack.

**Comment by Windblade-GR01** [comment]
`engine.ini`'s `Screen-type = dynamic / resizable`, doesn't work either.

**Comment by Wohlstand** [comment]
Yeah, the sense of `screen-type` is choosing the type of in-game scene: let it be fixed-size like it does now, or will be scalable in a dependence on a window size. The second case will don't scale textures and drawing objects, but will allow to change of camera size that will see same square (in pixels^2) as the physical square of the window. (For High-DPI mode, I'll scale the stuff to avoid the crap that PGE Engine had some times ago on macOS).

**Comment by Windblade-GR01** [comment]
Not sure how this would work under `Screen-type = dynamic / resizable`.
The rooms could scale up to `3840 x 2160` (4K resolution)
For now the only option looks to have the ability to limit the maximum dynamic resolution.
Even 2d sandbox games have this problem.

This forum link have images of starbound (A game like terraria) with absurd levels of zoom
[https://community.playstarbound.com/threads/guide-how-to-zoom-out-more-than-game-settings-allow.43009/](url)
It uses a dynamic resolution like you plan with Moondust and you can see at high de-zoom or ultra high resolutions it breaks.

If you limit the maximum resolution to the section size and/or a max-height max-width you could prevent the problem showed you adobe (The links).

PD: More images of how it looks.
[https://imgur.com/a/44QS5](url)
[http://i.imgur.com/mNWTF15.jpg](url)

**Comment by Windblade-GR01** [comment]
TL;DR: To make `Screen-type = dynamic / resizable` playable at high resolutions the Moondust team needs to limit the scaling to not exceed a maximum resolution setting (This varies per game) and to stay in the room margins.

Example
Maximum resolution = 960 x 540
Section size = 480 x 270
Resolution = 480 x 270 scaled to 960 x 540

Example 2
Maximum resolution = 960 x 540
Section size = 4200 x 8500
Resolution = 960 x 540

**Comment by Wohlstand** [comment]
You can don't worry to the room size limit as I have made the black zone and alignment to center that will allow to have rooms smaller than screen size, already :wink:
The only a limit is minimal screen size to don't squash the thing into too small thing.

**Comment by Wohlstand** [comment]
On a quick hand, I have added the support for `screen-width` and `screen-height` values, and they are now working. It's the test I have ran with 1920x1080 in-game resolution (my screen size is two monitors of 1280x1024, therefore image is squashed):
![2018-11-10 02-24-00](https://user-images.githubusercontent.com/6751442/48293385-d0d6e280-e48f-11e8-87e5-2db6d1aef099.png)

And the 256x224 doubled into 2 to make the resolution be like one of SNES's:
![2018-11-10 02-27-46](https://user-images.githubusercontent.com/6751442/48293491-6c685300-e490-11e8-858a-d0c39538b26b.png)
![2018-11-10 02-27-53](https://user-images.githubusercontent.com/6751442/48293493-6ecaad00-e490-11e8-8757-983ee1ed53e4.png)
![2018-11-10 02-28-42](https://user-images.githubusercontent.com/6751442/48293494-70947080-e490-11e8-9ec1-012aa069e98e.png)

The dynamical resolutions didn't implemented yet, the statical only.

**Comment by Wohlstand** [comment]
But, be careful! Without of changing of world map setup, the view will suck on a non-standard resolution. So, it's suggested to take world map area in a full screen view and align HUD at top (most of those setup also can be done through `engine.ini`).

**Comment by Shadowblitz16** [comment]
wait does this unload/reset enemies outside of the screen boundaries? 
also would lunalua be able to support this?

**Comment by Wohlstand** [comment]
> wait does this unload/reset enemies outside of the screen boundaries?

NPCs are activating once they are appearing on the visible area. Once this visible area was increased, there are more NPCs will appear on the screen and will be activated, or the opposite: when you decreased the size of screen, less NPCs will be activated.

**Comment by Wohlstand** [comment]
> also would lunalua be able to support this?

In Discord that was discussed and LunaLua can't yet
